docs/en/sql-reference/functions/type-conversion-functions.md

```diff
@@ -1509,10 +1509,12 @@ parseDateTimeBestEffort(time_string [, time_zone])
 - A string containing 9..10 digit [unix timestamp](https://en.wikipedia.org/wiki/Unix_time).
 - A string with a date and a time component: `YYYYMMDDhhmmss`, `DD/MM/YYYY hh:mm:ss`, `DD-MM-YY hh:mm`, `YYYY-MM-DD hh:mm:ss`, etc.
 - A string with a date, but no time component: `YYYY`, `YYYYMM`, `YYYY*MM`, `DD/MM/YYYY`, `DD-MM-YY` etc.
-- A string with a day and time: `DD`, `DD hh`, `DD hh:mm`. In this case `YYYY-MM` are substituted as `2000-01`.
+- A string with a day and time: `DD`, `DD hh`, `DD hh:mm`. In this case `MM` is substituted by `01`.
 - A string that includes the date and time along with time zone offset information: `YYYY-MM-DD hh:mm:ss ±h:mm`, etc. For example, `2020-12-12 17:36:00 -5:00`.
+- A string that includes the date and time in the [syslog timestamp](https://datatracker.ietf.org/doc/html/rfc3164) format. For example, `Jun  9 14:20:32`.
 
 For all of the formats with separator the function parses months names expressed by their full name or by the first three letters of a month name. Examples: `24/DEC/18`, `24-Dec-18`, `01-September-2018`.
+If the year is not specified, it is considered to be equal to the current year. If the resulting date and time happens to be ahead of the current moment even by a second, the current year is substituted by the previous one.
```

Comment:
As far as I see, this behavior did not exist before, right? Is it really needed and does it break parsing of existing timestamps (i.e. is backward incompatible)? Would you like to add a test case for it (just parse a timestamp with a date very far in the future).

Reply:
That's right, this is new behavior. The missing year in all appropriate cases (not only syslog timestamp) used to be [substituted](https://github.com/ClickHouse/ClickHouse/pull/50925/commits/6d25e5a0d75325ddeee2be0d689da6ea4395fccc#diff-5de8fd1d7601ca60769a39870b2a7732b458a849c34f21115edccbfd33079676) by the hard-coded constant `2000`, so I might be backward incompatible - if previously someone happened to rely on it.
The [adjusted](https://github.com/ClickHouse/ClickHouse/pull/50925/commits/f096bfcad29b95ea02f720f0becd5931c5b7eb37) tests clearly demonstrate the changes made - I put the year `2000` in the arguments explicitly.
Could you please elaborate on the test to be added? Should I parse something like `Dec 31 59:59:59` in it?

Reply:
I see. The reason I asked is that there were recently some issues where some database behavior unexpectedly changed after an upgrade in an incompatible way and there was no way to switch back to the old behavior. In this case, a compatibility setting to toggle the behavior would have been useful.

On the other hand, if some behavior is not officially documented then the database is free to change it, at least from a "legal" perspective. Because the docs were silent about missing years, and because `parseDateTimeBestEffort` is a mess anyways (it is doing way too much, users should rather use `parseDateTime`), I think it is okay to omit a compatibility setting in this case.

About a new test for a string with missing year: Yes, it would need to asymptotically approach new year evening (Dec 31 59:59:59). There is a tiny chance that the test will fail once a year, but during this season, chances are high that most people are out of the office and don't care :-)

Reply:
Got it, thanks :)
Such test would work until the last second of this year, resulting in `2022-12-31 23:59:59` and after that `2023-12-31 23:59:59` forever, but I'll try apply some kind of the same "time freeze" magic in it.

Reply:
Done with the first part, still contemplating the test.